src/components/tableview/TableTable.vue

```diff
@@ -427,25 +442,62 @@ export default {
       }
       this.$set(this.pendingEdits, key, payload)
     },
+    addRowToPendingDeletes(row) {
+      row.getElement().classList.add('deleted')
+      const pkCell = row.getCells().find(c => c.getField() === this.primaryKey)
+
+      if (!pkCell) {
+        this.$noty.error("Can't delete row -- couldn't figure out primary key")       
+        return
+      }
+
+      const payload = {
+        table: this.table.name,
+        row: row,
+        schema: this.table.schema,
+        pkColumn: this.primaryKey,
+        primaryKey: pkCell.getValue()
+      }
+
+      this.$set(this.pendingDeletes, pkCell.getValue(), payload)
+    },
     async saveChanges() {
       try {
-        // throw new Error("This is an error")
-        const newData = await this.connection.updateValues(this.pendingEditList)
-        const updateIncludedPK = this.pendingEditList.find(e => e.column === e.pkColumn)
-        if (updateIncludedPK) {
+        let replaceData = false
+
+        // Commit Deletes
+        if (this.pendingDeleteList.length > 0) {
```

Comment:
I think probably this should check if this row has been edited and if so, remove it from the pendingEdits list.

Currently if you edit a row, then mark for deletion it will delete it then try to edit it, which could have side effects.

Reply:
Yes absolutely. It's also part of the refactoring mentioned above. I'll add it to this PR shortly.